### PR TITLE
Attacker's guide to governance

### PR DESCRIPTION
**Pull request by Chris-Hibbert**

### CODE REVIEW

File: packages/governance/docs/AttackGuide [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# An Attacker's guide to Agoric Governance
```
Comment (erights) [inline packages/governance/docs/AttackGuide]: It looks like you intend this to be a Markdown document. Please rename the file to end in `.md` and github will offer to render it.

Comment (Chris-Hibbert) [reply packages/governance/docs/AttackGuide]: done

File: packages/governance/docs/AttackGuide.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+If the code in BallotCounter, Registrar, ContractGovernor has subtle mistakes, wrong results will obtain.
+
+# Produce a discrepancy between Terms and actions in BallotCounter or Registrar
+
+The voter's assurance that a particular ballot has the effect they expect arises in part because the `terms` in the BallotCounter, Registrar, etc. dictate how those classes will act. If the code is changed to get info from hidden parameters or to ignore some of the terms, voters will be misled.
+
+# Use a timer that is controlled by a party to the vote
+
+Everyone involved relies on the timers used to close voting being platform timers, but the timers aren't self-revealing. Participants should compare the timers to known platform-provided timers before relying on them.
```
Comment (erights) [inline packages/governance/docs/AttackGuide.md]: Link to https://github.com/Agoric/agoric-sdk/issues/3748 ?

File: packages/governance/docs/AttackGuide.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ * mistakes that might arise in governed contracts 
+
+The first two can be mitigated by validation functions that cross-check linkages.
+
+I'm not as sure how to address the third, except better documentation, and support for custom validation functions.
+
+
+# Get a voter facet you shouldn't have
+
+Every module that handles voter facets should handle them carefully and ensure they don't escape. If you
```
Comment (erights) [inline packages/governance/docs/AttackGuide.md]: Ends with sentence fragment.

**Review by erights — COMMENTED** [review]
This document is not yet well formed. But it is a great start! I love the idea of it, and we should do similar docs for other components of our system.

Please bring this into a well formed enough state to merge and then ask me for another review. Thanks!

File: packages/governance/docs/AttackGuide.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# An Attacker's guide to Agoric Governance
+
+This is an incomplete list of potential weak points that an attacker might want to focus on when looking for ways to violate the integrity of the governance system. It's here to help defenders, but "attacker's mindset" is a good way to focus attention for the defender. The list should correspond pretty closely to the set of assurances that the governance system aims to support.
+
+I see two main kinds of threats here:
```
[the remark this answers is not in the document]
Comment (Chris-Hibbert) [reply packages/governance/docs/AttackGuide.md]: oops. Done.

File: packages/governance/docs/AttackGuide.md (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/governance/docs/AttackGuide.md]: completed the thought.

File: packages/governance/docs/AttackGuide.md (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/governance/docs/AttackGuide.md]: done

**Review by Chris-Hibbert — COMMENTED** [review]
reformatted and improved the consistency and appearance.

**Review by erights — APPROVED** [review]
LGTM good start!

**Pull request merged**